File: backend/src/main/java/com/taller/ingenieria/api/service/PaymentService.java
package com.taller.ingenieria.api.service;

import com.stripe.exception.SignatureVerificationException;
import com.stripe.exception.StripeException;
import com.taller.ingenieria.api.dto.response.PaymentIntentResponseDTO;

public interface PaymentService {

    /**
     * Crea una Intención de Pago (Payment Intent) en Stripe.
     * Calcula el total del carrito del usuario y le pide a Stripe que prepare
     * una transacción por ese monto.
     *
     * @param cartId El ID del carrito para usuarios anónimos (opcional).
     * @return Un DTO que contiene el 'client_secret' necesario para el frontend.
     * @throws StripeException Si hay un error al comunicarse con la API de Stripe.
     */
    PaymentIntentResponseDTO createPaymentIntent(String cartId, Integer idDireccion) throws StripeException;


    /**
     * Procesa y verifica un evento de webhook entrante de Stripe.
     * Si el evento es un pago exitoso, procede a crear el pedido en el sistema.
     *
     * @param payload El cuerpo de la petición (el JSON del evento) enviado por Stripe.
     * @param sigHeader El valor de la cabecera 'Stripe-Signature' para la verificación.
     * @throws SignatureVerificationException si la firma del webhook es inválida.
     */
    void handleStripeEvent(String payload, String sigHeader) throws SignatureVerificationException;
}

File: frontend/src/api/paymentService.js
import apiClient from "./apiClient";

/**
 * Llama al backend para crear una Intención de Pago (Payment Intent) en Stripe.
 * @param {number} idDireccion - El ID de la dirección de envío seleccionada por el usuario.
 * @returns {Promise<object>} La respuesta de la API, que debe contener el clientSecret.
 */
const createPaymentIntent = async (idDireccion) => {
  try {
    const payload = { idDireccion };
    const response = await apiClient.post("/payment/create-intent", payload);
    return response.data;
  } catch (error) {
    // Si el backend devuelve un error de validación (ej. falta el teléfono), lo lanzamos.
    throw (
      error.response?.data || { message: "Error al crear la intención de pago" }
    );
  }
};

export const paymentService = {
  createPaymentIntent,
};
